prboom2/src/SDL/i_system.c

```diff
@@ -344,22 +344,38 @@ const char* I_GetTempDir(void)
 // cph - V.Aguilar (5/30/99) suggested return ~/.lxdoom/, creating
 //  if non-existant
 // cph 2006/07/23 - give prboom+ its own dir
-static const char prboom_dir[] = {"/.prboom-plus"}; // Mead rem extra slash 8/21/03
+static const char prboom_dir[] = {"prboom-plus"};
 
 const char *I_DoomExeDir(void)
 {
   static char *base;
+  struct stat data_dir;
+
   if (!base)        // cache multiple requests
     {
       char *home = getenv("HOME");
       size_t len = strlen(home);
+      size_t p_len = (len + strlen(prboom_dir) + 3);
 
-      base = malloc(len + strlen(prboom_dir) + 1);
-      strcpy(base, home);
       // I've had trouble with trailing slashes before...
-      if (base[len-1] == '/') base[len-1] = 0;
-      strcat(base, prboom_dir);
-      mkdir(base, S_IRUSR | S_IWUSR | S_IXUSR); // Make sure it exists
+      if (home[len-1] == '/') home[len-1] = 0;
```

Comment:
"The application shall ensure that it does not modify the string pointed to by the getenv() function."
https://pubs.opengroup.org/onlinepubs/009696899/functions/getenv.html

Reply:
Yikes, so that's why that was done.